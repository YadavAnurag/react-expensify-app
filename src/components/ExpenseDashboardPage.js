import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const ExpenseDashboardPage = () => (
  <div>
    <div>
      <ExpenseListFilters />
      <ExpensesSummary />
      <ExpenseList />
    </div>
  </div>
);

export default ExpenseDashboardPage;