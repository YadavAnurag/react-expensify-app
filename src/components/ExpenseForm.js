import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

export default class ExpenseForm extends React.Component{
  constructor(props){
    super(props);

    this.state = {
      description: props.expense ? props.expense.description : '',
      amount: props.expense ? (props.expense.amount / 100).toString() : '',
      note: props.expense ? props.expense.note : '',
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      calanderFocused: false,
      error: ''
    }
  }
  onDescriptionChange = (e)=>{
    const description = e.target.value;
    this.setState(()=>({ description }));
  };
  onAmountChange = (e)=>{
    const amount = e.target.value;
    
    if(!amount || amount.match(/^\d{1,}(\.?\d{0,2})$/)){
      this.setState(()=>({ amount }));
    }
  
  };
  onNoteChange = (e)=>{
    const note = e.target.value;
    this.setState(()=>({ note }));
  };
  onAddExpense = (e)=>{
    e.preventDefault();
  };
  onDateChange = (createdAt)=>{
    if(createdAt){
      this.setState(()=>({ createdAt }));
    }
  };
  onFocusChange = ({ focused })=>{
    this.setState(()=>({ calanderFocused: focused }));
  };
  onSubmit = (e)=>{
    e.preventDefault();
    
    if(!this.state.amount || !this.state.description){
      this.setState(()=>({ error: 'Please enter amount and description' }));
    }else{
      this.setState(()=>({ error: '' }));

      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  };

  render(){
    return (
      <div>
        <div>{ this.state.error && <p>{this.state.error}</p> }</div>
        <form onSubmit={this.onSubmit}>
          <input 
            type="text"
            placeholder='description'
            value={this.state.description}
            onChange={this.onDescriptionChange}
            autoFocus
          />
          <input 
            type='text'
            placeholder='amount'
            value={this.state.amount}
            onChange={this.onAmountChange}
          />
          <SingleDatePicker
            date={this.state.createdAt}
            focused={this.state.calanderFocused}
            onDateChange={this.onDateChange}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={(day)=>(false)}
          />
          <textarea 
            name="note"
            placeholder='note'
            value={this.state.note}
            onChange={this.onNoteChange}
          ></textarea>
          <button>Add Expense</button>
        </form>
      </div>
    );
  };
}
