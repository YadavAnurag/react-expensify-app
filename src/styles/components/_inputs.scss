.text-input{
  border: 1px solid #cacccd;
  border-radius: 3px;
  height: 50px;
  font-size: $font-size-large;
  font-weight: 300;
  padding: $s-size;
}

.select{
  @extend .text-input;
}

.textarea{
  @extend .text-input;
  font-size: $font-size-small;
}