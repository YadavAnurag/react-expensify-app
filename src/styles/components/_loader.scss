.loader{
  align-items: center;
  background: #e5eff1;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100vw;
}
.loader__image{
  height: 8rem;
  width: 8rem;
}