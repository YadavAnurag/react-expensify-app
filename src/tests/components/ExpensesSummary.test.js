import { ExpensesSummary } from '../../components/ExpensesSummary';
import React from 'react';
import { shallow } from 'enzyme';

test('should render Expneses Summary with 1 expense correctly', ()=>{
  const wrapper = shallow(<ExpensesSummary  expenseCount={1} expenseTotal={100} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render Expneses Summary with multiple expenses correctly', ()=>{
  const wrapper = shallow(<ExpensesSummary expenseCount={3} expenseTotal={200} />);
  expect(wrapper).toMatchSnapshot();
});