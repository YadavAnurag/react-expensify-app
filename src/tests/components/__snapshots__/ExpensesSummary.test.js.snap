// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expneses Summary with 1 expense correctly 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    $1.00
  </h2>
</div>
`;

exports[`should render Expneses Summary with multiple expenses correctly 1`] = `
<div>
  <h2>
    Viewing 
    3
     
    expneses
     totalling 
    $2.00
  </h2>
</div>
`;
